### ISSUE TITLE
Writer.new(): KeyError: 'CN'

### ISSUE THREAD

**Issue by t0neg** [cannatag/ldap3#256]
Hello,

ldap3 version: 2.1.0
LDAP Server: Microsoft Active Directory

I'm trying to create new entry using Writer class.
```
server = Server('192.168.0.11', get_info=ALL)
conn = Connection(server,
                  'CN=Service Active Directory,OU=Service Accounts,OU=Company OU,DC=company,DC=int',
                  '<password>',
                  auto_bind=True)

obj_user = ObjectDef(object_class='user',
                     schema=conn)

w = Writer(conn, obj_user)
w.new('CN=Test User,OU=Test,DC=company,DC=int')
```

Stack Trace:
```
Traceback (most recent call last):
  File "D:/temp/ldap3_test.py", line 14, in <module>
    w.new('CN=Test User,OU=Test,DC=osc,DC=int')
  File "D:\python_virtualenv\ve_tls_ldap_utils\lib\site-packages\ldap3\abstract\cursor.py", line 673, in new
    entry.__dict__[rdn[0]].set(rdn[1])
KeyError: 'CN'
```

entry.entry_attributes = <class 'list'>: ['cn', 'objectClass', 'objectCategory', 'nTSecurityDescriptor', 'instanceType']
rdn = <class 'tuple'>: ('CN', 'Test User')

I guess case should be converted somewhere. What would it be more appropriate place to manage it?  At ldap3.utils.dn.safe_dn or directly at Writer.new()?

**Comment by cannatag** [comment]
This should work as intended. Let me check what's wrong with the case folding.

**Comment by cannatag** [comment]
Can you check code in dev? It should work now.

**Comment by t0neg** [comment]
Now it works perfectly. Thanks for fast fix!

**Comment by cannatag** [comment]
Released in 2.1.1
